controllers/release/adapter_test.go

```diff
@@ -935,6 +872,170 @@ var _ = Describe("Release Adapter", Ordered, func() {
+			Expect(adapter.registerDeploymentStatus(nil)).To(Succeed())
+			Expect(adapter.release.IsDeploying()).To(BeFalse())
+		})
+
+		It("does nothing if the binding doesn't have the expected condition", func() {
+			Expect(adapter.registerDeploymentStatus(snapshotEnvironmentBinding)).To(Succeed())
+			Expect(adapter.release.IsDeploying()).To(BeFalse())
+		})
+
+		It("registers the deployment status when the binding has been deployed", func() {
+			adapter.release.MarkDeploying("")
+			newSnapshotEnvironmentBinding := snapshotEnvironmentBinding.DeepCopy()
+			newSnapshotEnvironmentBinding.Status.ComponentDeploymentConditions = []metav1.Condition{
+				{
+					Type:   applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed,
+					Status: metav1.ConditionTrue,
+					Reason: "Deployed",
+				},
+			}
+			Expect(adapter.registerDeploymentStatus(newSnapshotEnvironmentBinding)).To(Succeed())
+			Expect(adapter.release.HasDeploymentFinished()).To(BeTrue())
+		})
+	})
+
+	Context("When registerProcessingData is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+		})
+
+		It("does nothing if there is no PipelineRun", func() {
+			Expect(adapter.registerProcessingData(nil, nil)).To(Succeed())
+			Expect(adapter.release.Status.Processing.PipelineRun).To(BeEmpty())
+		})
+
+		It("does nothing if there is no ReleaseStrategy", func() {
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pipeline-run",
+					Namespace: "default",
+				},
+			}
+			Expect(adapter.registerProcessingData(pipelineRun, nil)).To(Succeed())
+			Expect(adapter.release.Status.Processing.PipelineRun).To(BeEmpty())
+		})
+
+		It("registers the Release processing data", func() {
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "pipeline-run",
+					Namespace: "default",
+				},
+			}
+			Expect(adapter.registerProcessingData(pipelineRun, releaseStrategy)).To(Succeed())
+			Expect(adapter.release.Status.Processing.PipelineRun).To(Equal(fmt.Sprintf("%s%c%s",
+				pipelineRun.Namespace, types.Separator, pipelineRun.Name)))
+			Expect(adapter.release.Status.Processing.ReleaseStrategy).To(Equal(fmt.Sprintf("%s%c%s",
+				releaseStrategy.Namespace, types.Separator, releaseStrategy.Name)))
+			Expect(adapter.release.Status.Target).To(Equal(pipelineRun.Namespace))
+			Expect(adapter.release.IsProcessing()).To(BeTrue())
+		})
+	})
+
+	Context("When registerProcessingStatus is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+		})
+
+		It("does nothing if there is no PipelineRun", func() {
+			Expect(adapter.registerProcessingStatus(nil)).To(Succeed())
+			Expect(adapter.release.Status.CompletionTime).To(BeNil())
```

Comment:
shouldn't this be checking the processing completion time?

Reply:
Indeed.